pkg/ARCHITECTURE.md

```diff
@@ -0,0 +1,43 @@
+# Backend code structure
+
+Grafanas backend is written in GO using sqlite3/mysql or postgres as storage for dashboards, users etc. When Grafana was born there didnt exist much guides or direction for how to write medium sized application. So there are parts of Grafana code base that didn't quite pan out as we wanted. More about that under current rewrites! :)
+
+## Central folders of Grafanas backend
+
+| folder | description | 
+| ------- | ----------- |
+| /pkg/api | Http Handlers and routing. Almost all handler funcs are global which is something we would like to improve in the future. Handlers should be assosicated with a struct that refers to all depedencies. Ex bus |
+| /pkg/cmd | The binaries that we build. Grafana-server and grafana-cli |
+| /pkg/components | Mixed content of packages that we copied into Grafana and packages we implemented ourself. The Purpose of this folders should be packages that are too big for util and doesnt belong somewhere else |
+| /pkg/infra | Packages in infra should be packages that are used in multiple places of Grafana without knowing anything about Grafana domain. E.g. Logs, Metrics, Traces |
+| /pkg/services | Packages in services are responsible for peristing domain objects and manage the relationship between domain objects. |
```

Comment:
What tripped me with the `services` is that there is some modules that use DI and some that do not. Maybe we can mentioned that here?